### ISSUE TITLE
TypeHintDeclarationSniff incorrectly reports about missing typehint when parameter docBlock contains union of two types, one is traversable and the second one does not specify the traversable type

### ISSUE THREAD

**Issue by maryo** [slevomat/coding-standard#126]
```
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
 273 | ERROR | [x] Method \Vanio\Stdlib\Uri::withQuery() does not
     |       |     have parameter type hint for its parameter $query
     |       |     but it should be possible to add it based on
     |       |     @param annotation "string|array".
```

I think it should report only when the annotation is something like "array|string[]". Am I right?
I guess you would probably typed it like `string|mixed[]` and then it would be probably quite OK but I don't use this notation for mixed arrays. I've excluded "MissingTraversableParameterTypeHintSpecification, MissingTraversablePropertyTypeHintSpecification and MissingTraversableReturnTypeHintSpecification" but still consider this sniff very handy.

BTW thanks for open sourcing this repo.

**Comment by kukulich** [comment]
@maryo Could you post here the code of the method please?

**Comment by kukulich** [comment]
Fixed in 2bcf4016c91d307e65e9ce8f855a23262e66c622
